CMakeLists.txt

```diff
@@ -69,6 +69,25 @@ if ("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_BINARY_DIR}")
 endif()
 
 ################################################################################
+# Sanity Check: Check for in source build of legacy build system.
+# Some build files from the old build system could interfere with our build.
+################################################################################
+set(KLEE_LEGACY_BUILD_FILES
+  "include/klee/Config/config.h"
+  "test/lit.site.cfg"
+)
+foreach (legacy_file ${KLEE_LEGACY_BUILD_FILES})
+  if (EXISTS "${CMAKE_SOURCE_DIR}/${legacy_file}")
+    message(FATAL_ERROR "\"${CMAKE_SOURCE_DIR}/${legacy_file}\""
+      " exists in KLEE's source tree. It is likely that the legacy build system"
+      " was configured to do an in-source build in KLEE's source tree. This"
+      " could cause problems with the build so you must remove old build files"
+      " and then run cmake again."
```

Comment:
Can you change the message to say exactly what one should do.  Remove old build files is too vague.